### DISCUSSION TITLE
How to load GLTF or GLB from disk ?

### DISCUSSION THREAD

**Discussion by agGitHub** [sceneview/sceneform-android#226] [Q&A] [answered]
Hi,

I have GLTF and GLB files generated at runtime. I want to load them at runtime. 

When a file is loaded from the bundle, the code is like this:

```
CompletableFuture<ModelRenderable> model = ModelRenderable
                .builder()
                .setSource(this
                        ,  Uri.parse("models/textured.gltf") )
                .setIsFilamentGltf(true)
                .setAsyncLoadEnabled(true)
                .build();
```

How should I set the setSource (what URL ?) if I want to load a file from disk (from /storage/emulated/0/Android/data/com.my.project/files/textured.gltf ) ?

Thanks.

**Comment by ThomasGorisse** [comment]
Hi,

Please use discussion for this.

**Comment by grassydragon** [comment, marked as answer]
Hi!
You can try using the file URI: https://stackoverflow.com/questions/3004713/get-content-uri-from-file-path-in-android

**Comment by agGitHub** [comment]
Thanks. Actually, it is just my URL that was wrong. 

It is:
/storage/emulated/0/Android/data/com.my.project/files/models/textured.gltf 

and not:
/storage/emulated/0/Android/data/com.my.project/files/textured.gltf 

Best.
